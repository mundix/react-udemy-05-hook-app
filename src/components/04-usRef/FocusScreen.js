import React, { useRef } from 'react';
import '../02-useEffect/effects.css';

export const FocusScreen = () => {

    const inputRef = useRef();

    const handleClick = () => {
        console.log(inputRef); //return { current:undefined }
        inputRef.current.select(); //se usa la propiedad current 
    }

    return (
        <div>
            <h1>Focus Screen</h1>
            <hr />

            <input
                ref={inputRef} 
                type="text" 
                className="form-control" 
                placeholder="Su Nombre" 
                />
            <button className="btn btn-outline-primary mt-5" onClick={handleClick}>
                Focus
            </button>

        </div>
    )
}
