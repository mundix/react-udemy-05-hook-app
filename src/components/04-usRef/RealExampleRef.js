import React, { useState } from 'react';
import '../02-useEffect/effects.css';
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';

export const RealExampleRef = () => {

    const [show, setShow] = useState(false)

    return (
        <div>
            <h1>RealExampleRef</h1>
            <hr />
            {show && <p>existe</p>}
            { show && <MultipleCustomHooks/>}
            
            <button className="btn btn-success mt-5"
                onClick={()=> {
                    setShow(!show);
                }}
            >
                Show/Hide
            </button>

        </div>
    )
}
