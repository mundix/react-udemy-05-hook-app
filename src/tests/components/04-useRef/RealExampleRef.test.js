import React from 'react';
import { shallow } from 'enzyme';
import { RealExampleRef } from '../../../components/04-usRef/RealExampleRef';

describe('Pruebas en <RealExampleRef/>', () => {
    const wrapper = shallow(<RealExampleRef/>);
    
    test('Debe mostrar el <RealExampleRef/>', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('RealExampleRef').exists()).toBe(false);

    })

    test('Debe mostrar el componente <MultipleCustomHooks/>', () => {
        wrapper.find('button').simulate('click');
        expect(wrapper).toMatchSnapshot();
        console.log(wrapper.html());
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);    
        
    })
    
});
