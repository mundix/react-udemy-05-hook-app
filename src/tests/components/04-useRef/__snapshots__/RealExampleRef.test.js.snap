// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RealExampleRef/> Debe mostrar el <RealExampleRef/> 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-success mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;

exports[`Pruebas en <RealExampleRef/> Debe mostrar el componente <MultipleCustomHooks/> 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <p>
    existe
  </p>
  <MultipleCustomHooks />
  <button
    className="btn btn-success mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
